Adds toggleable darkmode to the game!

Kmc2000 opened the pull request:
## About The Pull Request

Good morning /tg/, this PR adds dark-theme to replace the aging white design. You can use the "toggle darkmode" verb to opt-out of this change.

## Why It's Good For The Game

White theme still default.

This involves changing a few colours to match both light and dark mode, one of them being the OOC colour.

Demo:
https://gyazo.com/49c8a4637e81a654326b9a4aa5c05c38

WHY SO MUCH WINSET

If you can think of a better way please review, but I asked Lummox and he suggested winset() calls.

If you change ANY PART OF THE SKIN you will need to update the winset calls here too!

Why is darktheme on by default?
This PR is a port of my own work at:
https://github.com/DDMers/Deep-Space-13/pull/32

Where I decided to make it a default. If you object then I can change this (though it'll be a hassle)

EDIT: Thanks for the reddit gold kind stranger!

## Changelog
:cl: Kmc2000
add: Added darkmode! You can opt-in to this by clicking the settings icon on goonchat, and looking for the toggle darkmode button.
/:cl:

Kmc2000 committed:
Adds toggleable darkmode to the game! (Use the toggle darkmode verb if you want to use light mode)

Kmc2000 committed:
Removes a random linechange

Kmc2000 committed:
Undo unintentional changes

optimumtact commented:
you probably can't flip the default on people like that.

Kmc2000 commented:
Would you prefer I just make the pref non-default, or fiddle around with changing the skin again?

optimumtact commented:
ah to hell with it, we'll just go with it

Kmc2000 commented:
👍

Kmc2000 commented:
Double checking any colours that don't fit now

SpaceManiac commented on code/modules/client/preferences_toggles.dm:
```diff
@@ -454,3 +454,12 @@ GLOBAL_LIST_INIT(ghost_orbits, list(GHOST_ORBIT_CIRCLE,GHOST_ORBIT_TRIANGLE,GHOS
+	set desc = "Change your chat theme"
+	prefs.toggles ^= DARKMODE
+	prefs.save_preferences()
+	to_chat(usr, "Dark theme (YOU MUST RECONNECT TO SEE THIS CHANGE) [(usr.client.prefs.toggles & DARKMODE) ? "Enabled" : "Disabled"]")
```
What's up with this? Can't you just pile on the winsets?

Kmc2000 replied:
Yeah it's more so goonchat reloads as I didn't want to break it by loading it again.

Kmc2000 replied:
I guess I could delete their browseroutput and make a new one?

SpaceManiac replied:
Rather than sending one of light/dark CSS as the single css file to goonchat, send both and add some way to tell goonchat to switch which is active.

Kmc2000 replied:
Again, still requires goonchat to reload no?

SpaceManiac replied:
should be possible to do without a reload

Kmc2000 replied:
But wouldn't the client still have the wrong assets? Or would the new ones override the old.

SpaceManiac replied:
Send both assets always, instruct goonchat to change which one is being used

Kmc2000 replied:
Ohhhhhhh I'll give that a shot.

Kmc2000 replied:
Actually that still won't help. If they go from dark -> white they'll still have the wingets applied and must relog anyway.

SpaceManiac replied:
Second function which `winset`s everything the other way

Kmc2000 replied:
I know 🤔 
That just seems like a lot of extra spaghetti for quite a small issue /shrug

SpaceManiac replied:
Maybe personal preference, but forcing users to relog for no good reason is a huge red flag that a feature is not well-thought-ought IMO. In the past, changing UI style only applied on relog, so it basically just gave off the appearance that that preference didn't do anything. #38089 involved refactoring to fix it but now the feature actually works rather than being confusing.

optimumtact replied:
Mmm, but surely we can fix it with some explicit messaging about needing to log out and back in.


I'm sympathetic to the concern of having to have two winset functions that need to be kept up to date with style changes.

Kmc2000 replied:
My main concern is that, yeah. As there's no way to automate this every ui update means going through two sets of winsets which is horrible...

ninjanomnom replied:
SpaceManiac is right here.

You could just make the different modes be defined by a pair of datums and have winsets able to apply the whole thing from those

Kmc2000 replied:
That still requires manually replacing every window though...

If you can find someone willing to maintain that I'll do it.

Kmc2000 committed:
Changes some forced font tags to use spans, which won't break darkmode.

Rowelll commented:
Awesome.

MCterra10 commented:
Who cares about medical overhauls, THIS is what we needed

francinum commented on code/__DEFINES/misc.dm:
```diff
@@ -233,7 +233,7 @@ GLOBAL_LIST_INIT(ghost_others_options, list(GHOST_OTHERS_SIMPLE, GHOST_OTHERS_DE
 GLOBAL_LIST_INIT(pda_styles, list(MONO, VT, ORBITRON, SHARE))
 
 //Color Defines
-#define OOC_COLOR  "#002eb8"
```
Unstated change.

PKPenguin321 commented:
people are gonna be freaked out by this for a good while lol

Kmc2000 committed:
Fixes up some problems I found while testing
Darkmode is no longer default (this caused some issues with people getting a broken chat)

Kmc2000 commented:
I had to change it to not be on default for the time being, encountered some interesting issues when testing with other people who somehow got broken chats.

Bawhoppen commented:
I'd prefer it not to be the default please.

Dorsisdwarf commented:
I'm pretty sure that web design standards say that darkmode shouldn't be the default (Discord gets a free pass because their light mode is gross), so I'm not sure about making the new option default

MCterra10 commented:
Idea: make our light mode horrific so that we can make the default dark mode

epochayur commented:
sexy af

Anonymous-Player commented:
Did not even bother showing the common chat, attacks, various departmental channels. Make another one.

actioninja commented:
> I'm pretty sure that web design standards say that darkmode shouldn't be the default (Discord gets a free pass because their light mode is gross), so I'm not sure about making the new option default

Good thing this isn't web design.

Anonymous-Player commented:
You're right It isn't web design. He probably meant to say UI/UX standards instead of web design standards. The demo still doesn't show the colors of attacks, common chat, etc. It would be worth showing those in another demo so that people could request a font color change if there was an agreement for that particular aspect.

AffectedArc07 commented:
Doesnt this count as a port?
*ahem* https://github.com/ParadiseSS13/Paradise/pull/10758

Kmc2000 commented:
I've finished up with some code improvements on my downstream copy, will add them shortly.

Kmc2000 committed:
Darkmode sucks less now

Kmc2000 committed:
Darkmode button now works again

ShizCalev commented on code/__DEFINES/misc.dm:
```diff
@@ -233,7 +233,7 @@ GLOBAL_LIST_INIT(ghost_others_options, list(GHOST_OTHERS_SIMPLE, GHOST_OTHERS_DE
 
 //Color Defines
-#define OOC_COLOR  "#002eb8"
+#define OOC_COLOR  "#386aff"
```
revert

bobbahbrown commented:
the changelog says "opt-in" but in reality it's "opt-out" i can't believe this discrepancy exists this is the true state of the codebase

plapatin commented:
REVERT IT OH GOD WHAT THE FUCK IS THIS

Kmc2000 commented:
😎

MCterra10 commented:
Dude you don't need those shades now that it's merged

Kmc2000 commented:
🙃

Cyprex commented:
>White theme still default.

This is what the PR body claims, but that doesn't seem true.

Kmc2000 commented:
Yeah I fucked up copying some shit
It's in my fix pr.

I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE commented:
Is there any reason why you have two arguments per winset(), with both of them being the same property? E.g. "background-color = none;background-color = #2c2f33"
Seems like the second one just overwrites the first and removing the first one entirely makes no difference.
